cpp/src/copying/gather.hpp

```diff
@@ -58,12 +58,15 @@ namespace detail {
  * gather).
  * @param check_bounds Optionally perform bounds checking on the values of
  * `gather_map`
- * @param stream Optional CUDA stream on which to execute kernels
+ * @param merge_nvstring_category If set to true and both the source column and its
+ * corresponding destination column are of type `GDF_STRING_CATEGORY`, the
+ * `nvstring_category` objects of these will be merged before gather is performed. 
```

Comment:
Since the function is `sync_categories` lets stick with the "sync" terminology instead of "merge". 
```suggestion
 * `nvstring_category` objects of these will be synced before gather is performed. 
```